packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,1112 @@
+    splitAddress,
+    recipients,
+    distributorFeePercent,
+    totalAllocationPercent,
+    transactionOverrides = {},
+  }: UpdateSplitV2Config): Promise<TransactionFormat> {
+    const {
+      recipientAddresses,
+      recipientAllocations,
+      distributionIncentive,
+      totalAllocation,
+    } = getValidatedSplitV2Config(
+      recipients,
+      distributorFeePercent,
+      totalAllocationPercent,
+    )
+
+    validateAddress(splitAddress)
+    recipientAddresses.map((recipient) => validateAddress(recipient))
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    return this._executeContractFunction({
+      contractAddress: splitAddress,
+      contractAbi: splitV2ABI,
+      functionName: 'updateSplit',
+      functionArgs: [
+        {
+          recipients: recipientAddresses,
+          allocations: recipientAllocations,
+          totalAllocation,
+          distributionIncentive,
+        },
+      ],
+      transactionOverrides,
+    })
+  }
+
+  protected async _distribute({
+    splitAddress,
+    tokenAddress: token,
+    distributorAddress = this._walletClient?.account.address as Address,
+    transactionOverrides = {},
+  }: DistributeSplitConfig): Promise<TransactionFormat> {
+    validateAddress(splitAddress)
+    validateAddress(token)
+    validateAddress(distributorAddress)
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+    let split: Split
+
+    if (this._dataClient)
+      split = await this._dataClient.getSplitMetadata({
+        chainId: this._chainId,
+        splitAddress,
+      })
+    else split = (await this._getSplitMetadata(splitAddress)).split
+
+    const {
+      recipientAddresses,
+      recipientAllocations,
+      totalAllocation,
+      distributionIncentive,
+    } = getValidatedSplitV2Config(
+      split.recipients.map((recipient) => {
+        return {
+          address: recipient.recipient.address,
+          percentAllocation: recipient.percentAllocation,
+        }
+      }),
+      split.distributorFeePercent,
+    )
+
+    return this._executeContractFunction({
+      contractAddress: splitAddress,
+      contractAbi: splitV2ABI,
+      functionName: 'distribute',
+      functionArgs: [
+        {
+          recipients: recipientAddresses,
+          allocations: recipientAllocations,
+          totalAllocation,
+          distributionIncentive,
+        },
+        token,
+        distributorAddress,
+      ],
+      transactionOverrides,
+    })
+  }
+
+  async _paused(splitAddress: Address): Promise<boolean> {
+    this._requirePublicClient()
+    return this._getSplitV2Contract(splitAddress).read.paused()
+  }
+
+  async _owner(splitAddress: Address): Promise<Address> {
+    this._requirePublicClient()
+    return this._getSplitV2Contract(splitAddress).read.owner()
+  }
+
+  async _getSplitMetadata(splitAddress: Address): Promise<{ split: Split }> {
```

Comment:
nit: can we make the name slightly more clear? something like `_getSplitMetadataUsingProvider` or something?